### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by sabetAI** [raboof/dualfisheye2equirectangular#5]
I'm trying to generate remapping for the Theta S with 1280x720 resolution images from dual fish eye cameras. I'm getting a segmentation fault when I run:

`./projection -x xmap_thetaS.pgm -y ymap_thetaS.pgm -h 720 -w 1280 -r 640 -c 1280 -b 75 -m thetas`

which is confusing. I thought `projection` was supposed to _generate_ the `.pgm` files, not use pre-existing ones. What am I missing here?

**Comment by raboof** [comment]
You are correct projection should generated the `.pgm` files. Looks like a bug.

**Comment by sabetAI** [comment]
Any luck?

**Comment by raboof** [comment]
Sorry, I haven't had any free time to devote to this lately. Does running this command via `gdb` and looking at the backtrace reveal anything interesting?

**Comment by josephduchesne** [comment]
It's crashing on line 500 of the fork I've got:

`image_x[(cfg.height - 1) - y][x + cfg.cols / 2] =(int)round(o.y) + cfg.crop + cfg.width / 2;`

indices: 719, 640
x,y = 0,0
cfg.height = 720
cfg.cols = 1280

Works fine if you tweak the output size:
```
./projection -x xmap_thetaS.pgm -y ymap_thetaS.pgm -h 720 -w 1280 -r 720 -c 1440 -b 75 -m thetas
```

**Comment by raboof** [comment]
Would you be up for creating a PR to fix the issue?

**Comment by josephduchesne** [comment]
I'm afraid not at the moment, I was looking into this for work and changing the output parameters as a workaround was fine to let me proceed. I didn't get a chance to dig into how the function works enough to figure out why it's crashing (i.e.. what bounds checks are missing maybe).

**Comment by raboof** [comment]
Great that the workaround has unblocked you!

> I was looking into this for work

Allocating some time to fix the bug would be a great way to give back!
